src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -10,24 +10,36 @@ object RockSlicer {
   def main(args: Array[String]) {
     val conf = new SparkConf().setAppName("SparkRocks")
     val sc = new SparkContext(conf)
-    val arguments = CommandReader.parseArguments(args)
+    val parsedArgs = CommandReader.parseArguments(args)
+    if (parsedArgs.isEmpty) {
+      // Error message will already be printed out by CommandReader
+      System.exit(-1)
+    }
+    val arguments = CommandReader.parseArguments(args).get
 
     // Open and read input file specifying rock volume and joints
-    val inputSource = Source.fromFile(arguments('inputFile).toString)
-    val (rockVolume, joints) = InputProcessor.readInput(inputSource)
+    val inputSource = Source.fromFile(arguments.inputFile)
+    val inputOpt = InputProcessor.readInput(inputSource)
     inputSource.close()
-    var blocks = Vector(Block((0.0, 0.0, 0.0), rockVolume))
+    if (inputOpt.isEmpty) {
+      // Error message will already be printed out by InputProcessor
+      System.exit(-1)
+    }
+    val (rockVolume, joints) = inputOpt.get
+    var blocks = Seq(Block((0.0, 0.0, 0.0), rockVolume))
```

Comment:
Looking at this now, I realize it would be good idea to get the global origin coordinates from the input file and use it here when constructing `blocks`.
This will make a difference when compared to the data for `joints`, since those are specified with a global origin explicitly while it's only implicit in the faces of `blocks`. This will make sure that the distances are actually specified relative to the same origin. 
This is probably a left-over from when we initially just got things rolling. Does this make sense?